sys/include/event.h

```diff
@@ -149,21 +151,83 @@ typedef struct {
     thread_t *waiter;           /**< thread ownning event queue         */
 } event_queue_t;
 
+
+/**
+ * @brief   Initialize an array of event queues
+ *
+ * This will set the calling thread as owner of each queue in @p queues.
+ *
+ * @param[out]  queues      event queue objects to initialize
+ * @param[in]   n_queues    number of queues in @p queues
+ */
+static inline void event_queues_init(event_queue_t *queues,
+                                          size_t n_queues)
+{
+    assert(queues && n_queues);
+    thread_t *me = (thread_t *)sched_active_thread;
+    for (size_t i = 0; i < n_queues; i++) {
+        memset(&queues[i], '\0', sizeof(queues[0]));
```

Comment:
can this be moved out of the loop:

```memset(queues, '\0', sizeof(queues[0]) * n_queues);```

?

Reply:
This yields [slightly more instructions when moved out of the loop](https://godbolt.org/#z:OYLghAFBqd5TKALEBjA9gEwKYFFMCWALugE4A0BIEAZugHZEDKqAhgDbYgCMALOQCse5dq3qgA%2BlNTkAztk6oiBBtWz1MAYXTsArgFt6IAEzl1mADIF62AHIGARtlImAbOQAO6WcRX1teoYmnt6%2BDFY29vpOLsbu8orKDExErKREAQZGpgnYSn4paUQRdo7ObnKp6ZlBOVXF1qXR5XEAlHLouqSoXADkAKTGAMzWqHo4ANT9Q5qyRISMAHRI07j9AAwAgoMj9GO6k9Oz8zg0y6sb28Oj49hTM3Ok1sDnQ2tbl0QAnh7YpxOPXRKKYAdgAQpcJlCJtYiBNMAZ9ARsLJ%2BgBWMHcdbogAi0whHxBOImRCQpGwrEwEiI%2BMunx%2Bf2wNABRFIQLh7AIcwk9Cwd364Mh0MBwM53N5kwAVDYAB40oYE7ZEiZiog8vnU2kfLbfX7/VXqnDUiZjLlqiXYTUKuk6hn/EVwgWK6Emg0W43YABu6jVqq1mxdpPJlONkoA7qxiM5/QLiV6fRIAI66bApq2Ky6e9AETAw%2BjECSetLcCDxxhJlNpuGS5OplHkAEEABelrh9ArddkrSdQqh2DlznoJLJFKp1f02H9LropAmEB8LeNBHuxOxCph900E3btZTqPXBEGEOMYNaoOdLuhE/08iIEEGrl3KPRYMPaJxDfWDYX2HQNAgT6ohi2Lvq03bWlsl7QoBL5vjiiwRlGs7TMSE5TtCsZ0kSvTtOwIC9GivTkEYvTrER6D4VusidN0/LDNwRFEPhZFgeQADWIC8AAnIs3Boq4xhoiI%2BG8ER%2BggGiX4kWR5AUb0RGyCAX5MaROHkHAsAoBg%2BgeAQnAUFQEDabp%2BloGk%2Bggli5A0HpRDOIpEAOMxREONYaRfPhDHkNpE6MAA8vQ7Aeap5A4PoYjAJwzmhQQ5L5N6ikhf2eS6HZnlEbCCjRZyDikO52g4Ol5CsgQ4m9Ax7R0IwLAcFwfCCMIojiCAUgSDIOWKZA7ToB4ST0IlAC0fmyApCh5H1agaDURjcGYGglFEMTCF4Ph9dNy2hH1C1lC4s25PkyT1Ote1jQd9CFOk23NLtlRFMdt2XY0i3lNw7TUV0PQ8Lh%2BGEcR0VyZsABKACyEzAKgqATCCizGLxc68jYZ4QPgxBkFM9ENtoOl6c46NDK9EyaIxzmsUgo7lBA329KJ5DiZJf0hXJClKcVJPtBx3GLLwQwgkJeG9EMRHSeR%2BHE6p7QaQg8AQFpfIoyQBnUFVzBsFF9VCKYTWSNIcinRNpZTToWTCOYV1LbNK1hP4RtBBbm1%2BGbL264kBRHTbM3O%2BNrtFI7N1zHd7vCP7j2RDtX0dB9fQ7DcBz8g88ywq87xXLs%2ByHPHmCnEndLXHstybo8zzZ9qmy6oyzIOuevZ5nCCL6Eiz4YliuIxsqQajumNql3aTIsmyopmoa/KCpBwqsuyKqDxaEzSv28oZsqboavPXdl/qU/L66G9GivJdr73lc9qPUKmuKm9lr6ZroVC7chtWiF2aQrdxt65ZPp3JdZjmeYFkWpDGKWV%2Bap37VkAt%2BZsrZtwdj3N2EeAZoRz0HMOYMY4Z5oQgvAqE15bz3jiGAiYX5Gwtj/ABSsjcwQgRxGeSUUDALgQvFCGcc4fxLhXAQ/EG4ji0LIfuMEMIjxHjPEfTBl4YIYjgghSMj82HoIYaCPEhIcRqX5r9YWslKIAhoj0PGxgxYsXZhJdYix%2BKCWEtTMShiGYySZnIFmKkWLqUQDLNA6Bsb6UoNQYyOMXBsFIBZKyNl2CPwck5EKrl6DuSKj5H0AUgrRTChFKKIVCBxWUAlaKyVUCpT6F5TK/MZI5TyqQL4BUcmMSeGVCq1kGAq1qjwfgGsRARRajrDq8Buq9T8INYao0XaqANloQOs1TZPTDnbVafh7qWy2qM66Qc9be2qEMz2Z0LoNFDnMvabtAge2Duspo5s3paK4MYKmqj/r4SBqDcGkNoaw24PDBg2AkbyzRjsUwhNXEmVxu8s8RNWbi3aGTSkFMqY0zplJC58lbHKTZuxQxxiBJ83woLKxItoX2JwlTXRaL1EYrhd6UgPhVC8CAA).

I'm amazed that is actually result in `memset` being called at all. I would expect the compiler to use builtin implementation and optimize out the dead writes (e.g. not memsetting `waiter`). At least an unused `memset` (in which every member of a struct is initialized later on anyway) gets reliably optimized out.

Reply:
[Clang seems to optimize the call to `memset()` fully out, but only in case the `memset()` is part of the inner loop.](https://godbolt.org/z/YTjK7E) Very strange.

Reply:
Should I still move it out of the loop?

Reply:
nah, if it doesn't help in every case...